### PR TITLE
Fix broken windows build

### PR DESCRIPTION
**Pull request by andrewkho**
Windows build has been broken for a while on trunk, due to new MSVC deprecation warnings being treated by errors. It turns out the submodule pointer for aws-sdk-cpp hadn't been updated in a while. This PR updates it the pointer to 1.11.311, which can build on windows. Main branch does not build on MacOS due to a bug in cmake where it doesn't set the correct MACOSX_DEPLOYMENT_TARGET.

-
-

### CODE REVIEW

File: third_party/CMakeLists.txt [hunk 1]
```diff
@@ -54,6 +54,9 @@ else()
     else()
       list(APPEND AWSSDK_LIBRARIES ${aws_cpp_sdk_INSTALL}/lib/lib${lib}.a)
     endif()
+    if(MSVC)
+      target_compile_options(${lib} PRIVATE /W4 /WX)
```
Comment (divyanshk) [inline third_party/CMakeLists.txt:58]: Quick clarification, do we mean to make warnings surface better ? or suppress them ?

Comment (andrewkho) [reply third_party/CMakeLists.txt]: Ah sorry there's a compiler flag that makes warnings get treated as errors. In this case it's a deprecation warning and just trying to get it to ignore, not sure the right incantation

**Comment by ramanishsingh** [comment]
LGTM!

**Review by divyanshk — APPROVED** [review]
LGTM!

**Pull request merged**